docs/content/docs/ops/batch/recovery_from_job_master_failure.md

```diff
@@ -0,0 +1,99 @@
+  high availability (HA) is enabled. Flink supports HA services compatible with both ZooKeeper and Kubernetes.
+  For detailed configuration steps, please consult the section on [cluster high availability]({{< ref "docs/deployment/ha/overview#how-to-make-a-cluster-highly-available" >}})
+  in the Flink official documentation.
+- Configure [execution.batch.job-recovery.enabled]({{< ref "docs/deployment/config" >}}#execution-batch-job-recovery-enabled): true
+
+Note that currently only [Adaptive Batch Scheduler]({{< ref "docs/deployment/elastic_scaling" >}}#adaptive-batch-scheduler) 
+supports this feature. And Flink batch jobs will use this scheduler by default unless another scheduler is explicitly configured.
+
+### Configuration Optimization
+
+To enable batch jobs to recover as much progress as possible after a JobMaster failover, and avoid rerunning tasks 
+that have already been finished, you can configure the following options for optimization:
+
+- [execution.batch.job-recovery.snapshot.min-pause]({{< ref "docs/deployment/config" >}}#execution-batch-job-recovery-snapshot-min-pause):
+  This setting determines the minimum pause time allowed between snapshots for the OperatorCoordinator and ShuffleMaster.
+  This parameter could be adjusted based on the expected I/O load of your cluster and the tolerable amount of state regression. 
+  Reduce this interval if smaller state regressions are preferred and a higher I/O load is acceptable.
+- [execution.batch.job-recovery.previous-worker.recovery.timeout]({{< ref "docs/deployment/config" >}}#execution-batch-job-recovery-previous-worker-recovery-timeout):
+  This setting determines the timeout duration allowed for Shuffle workers to reconnect. During the recovery process, Flink 
+  requests the retained intermediate result data information from the Shuffle Master. If the timeout is reached, 
+  Flink will use all the acquired intermediate result data to recover the state.
+- [job-event.store.write-buffer.flush-interval]({{< ref "docs/deployment/config" >}}#job-event-store-write-buffer-flush-interval):
+  This setting determines the flush interval for the JobEventStore's write buffers.
+- [job-event.store.write-buffer.size]({{< ref "docs/deployment/config" >}}#job-event-store-write-buffer-size): This 
+  setting determines the write buffer size in the JobEventStore. When the buffer is full, its contents are flushed to the external
+  filesystem.
+
+## Limitations
+
+- Only compatible with the new Source (FLIP-27): Since the legacy Source has been deprecated, this feature is only 
+  compatible with the new Source.
+- Exclusive to the [Adaptive Batch Scheduler]({{< ref "docs/deployment/elastic_scaling" >}}#adaptive-batch-scheduler): 
+  Currently, only the Adaptive Batch Scheduler supports the recovery of batch jobs after a
+  JobMaster failover. As a result, the feature inherits all the
+  [limitations of the Adaptive Batch Scheduler]({{<ref "docs/deployment/elastic_scaling" >}}#limitations-2).
```

Comment:
IIRC, `SupportsBatchSnapshot` is also important for this feature to work better.